### PR TITLE
[UIPO-15] Product items carousel component

### PR DESCRIPTION
**Pull request by dimas0211**

### CODE REVIEW

**Comment by Hollister009** [comment]
By the way, if you are still working on PR. Name PR with [WIP] tag that can be removed after you are done with all changes

File: src/components/Carousel/Carousel.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+  checkIfSlidesAllTheWayOver() {
+    // if scrollLeft == 0
+    // hide left button
+    const allTHeWayLeftValue = (this.carouselViewport.current.scrollLeft === 0);
+
+    // if scrollLeft + viewPortOffset.length === whole viewPort length
+    // 9 cards - each 120px: 9 * 210all === whole viewPort length
+    // hide the rightScrollButton
+    const { scrollLeft, clientWidth, scrollWidth } = this.carouselViewport.current;
```
Comment (Nazar-bnm) [inline src/components/Carousel/Carousel.js]: move it to top and use `scrollLeft` for allTHeWayLeftValue

File: src/components/Carousel/Carousel.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    (allTheWayLeft !== allTHeWayLeftValue || allTheWayRight !== allTheWayRightValue) &&
+      this.setState({
+        allTheWayLeft: allTHeWayLeftValue,
+        allTheWayRight: allTheWayRightValue,
+      });
+  }
+
+  checkNumOfSlidesToScroll() {
+    const numOfSlidesToScroll = window.innerWidth <= 900 ? 2 : 4;
+    this.state.numOfSlidesToScroll !== numOfSlidesToScroll &&
```
Comment (Nazar-bnm) [inline src/components/Carousel/Carousel.js:88]: use object destructure for state

File: src/components/Carousel/Carousel.js [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    this.setState({ numOfSlidesToScroll });
+  }
+
+  handleClick(e) {
+    const clickedBtn = (e.currentTarget.classList.contains(`${CN}__left-nav`)) ? 'left' :'right';
+    const { numOfSlidesToScroll, widthOfSlide } = this.state;
+    const carouselViewport = this.carouselViewport.current;
+    const newPos = clickedBtn === 'left' ?
+      (carouselViewport.scrollLeft - (numOfSlidesToScroll * widthOfSlide)) :
```
Comment (Nazar-bnm) [inline src/components/Carousel/Carousel.js]: move `numOfSlidesToScroll * widthOfSlide` to const

File: src/components/Carousel/Carousel.js [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+      scrollDirection: 'scrollLeft',
+    });
+  }
+
+  render() {
+    const { children } = this.props;
+    const { allTheWayLeft, allTheWayRight } = this.state;
+    const navClasses = cx({
+      [`${CN}__nav`]: true,
+    });
```
Comment (Nazar-bnm) [inline src/components/Carousel/Carousel.js]: use 1 line for it

File: src/components/ProductList/ProductList.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    super(props);
+    this.state = { productList: [...products] };
+  }
+
+  render() {
+    const { productList } = this.state;
+
+    const listOfProducts =
+    productList &&
+    productList.map( ({ productName, category, images }) => (
```
Comment (Nazar-bnm) [inline src/components/ProductList/ProductList.js]: remove extra spaces

File: src/components/ProductList/ProductList.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+      <Slide
+        image = {images[0].url[0]}
+        name = {productName}
+        category = {category}
+        key = {productName}
```
Comment (Nazar-bnm) [inline src/components/ProductList/ProductList.js]: remove extra spaces

**Review by Nazar-bnm — CHANGES_REQUESTED** [review]

File: src/components/Carousel/Carousel.js (hunk 1 above)
Comment (dimas0211) [reply src/components/Carousel/Carousel.js]: done

File: src/components/Carousel/Carousel.js (hunk 2 above)
Comment (dimas0211) [reply src/components/Carousel/Carousel.js:88]: it does`n work with destructuring :(

File: src/components/Carousel/Carousel.js (hunk 3 above)
Comment (dimas0211) [reply src/components/Carousel/Carousel.js]: done

File: src/components/ProductList/ProductList.js (hunk 5 above)
Comment (dimas0211) [reply src/components/ProductList/ProductList.js]: done

File: src/components/ProductList/ProductList.js (hunk 6 above)
Comment (dimas0211) [reply src/components/ProductList/ProductList.js]: done

File: src/components/Carousel/Carousel.js (hunk 2 above)
Comment (Nazar-bnm) [reply src/components/Carousel/Carousel.js:88]: why ?

File: src/components/Carousel/Carousel.js [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+  }
+
+  onResize() {
+    this.resizeTheCarousel();
+    this.checkNumOfSlidesToScroll();
+  }
+
+  resizeTheCarousel() {
+    const { widthOfSlide } = this.state;
+    let carouselWidth = '0';
```
Comment (nadiiaLebedenko) [inline src/components/Carousel/Carousel.js:52]: You can assign `${widthOfSlide * 2}px` value here and remove the last 'else' statement

**Review by Nazar-bnm — APPROVED** [review]

File: src/components/Carousel/Carousel.js (hunk 7 above)
Comment (nadiiaLebedenko) [reply src/components/Carousel/Carousel.js:52]: @dimas0211 just FYI

**Pull request merged**